gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/util/DatetimeHelper.java

```diff
@@ -93,45 +97,41 @@ public static String format(final Instant d) {
      * </ul>>
      *
      */
-    public static Date parse(final String d) {
+    public static OffsetDateTime parse(final String d) {
```

Comment:
There's a lot going on in this method. Based on the javadoc comment, could it simply be replaced with [OffsetDateTime.parse(String)](https://docs.oracle.com/javase/8/docs/api/java/time/OffsetDateTime.html#parse-java.lang.CharSequence-) or [OffsetDateTime.from(TemporalAccessor)](https://docs.oracle.com/javase/8/docs/api/java/time/OffsetDateTime.html#from-java.time.temporal.TemporalAccessor-)?

Reply:
Not quite, dates without time or time without dates cannot be transformed into `OffsetDateTime`, so we'll still need the manual parsing. But you have a point that I can combine the last two cases.